src/plugins/data/public/autocomplete/autocomplete_service.ts

```diff
@@ -51,7 +51,7 @@ export class AutocompleteService {
   private getValueSuggestions?: ValueSuggestionsGetFn;
 
   private addQuerySuggestionProvider = (language: string, provider: QuerySuggestionGetFn): void => {
-    if (language && provider && this.autocompleteConfig.querySuggestions.enabled) {
+    if (language && provider && this.autocompleteConfig?.querySuggestions.enabled) {
```

Comment:
what led us to using optional chaining if we're not exercising this through testing?

Reply:
seems like we don't have any coverage on autocomplete_service.ts,

While writing unit tests for src/plugins/data/public/plugin, we initialize all the services, including

```
   this.searchService = new SearchService(initializerContext);
    this.uiService = new UiService(initializerContext);
    this.queryService = new QueryService();
    this.fieldFormatsService = new FieldFormatsService();
    this.autocomplete = new AutocompleteService(initializerContext);

```
The optional chaining ensures that if autocompleteConfig or querySuggestions is not defined (i.e., null or undefined), it won’t cause a runtime error when trying to access enabled.
In short, while the autocomplete_service.ts might not have direct test coverage, the use of optional chaining here is a safety feature that prevents errors when properties or objects are not available during execution. If the service were better covered in the tests, we’d be able to verify if this behavior is actually necessary or if there is a more predictable initialization pattern that could be tested directly.

Reply:
Did you add it unprompted as a safety measure, or because you saw it failing in some cases? I am curious about the other cases as well, but using this as an example.

Reply:
but why would querySuggestions be undefined? it's enforced by type and config-schema which provides a default value

Reply:
yes, i added it because the UTs were failing with `Cannot read property of undefined` for the following lines

Reply:
I would prefer for the UTs to be fixed to call the stack correctly. Do you know which ones were failing?